src/c2goto/library/builtin_libs.c

```diff
@@ -0,0 +1,24 @@
+/*******************************************************************\
+
+Module: Builtin Models
+
+Author: Rafael Menezes
+
+Description: Models for custom functions that are easier to implement
+ and test here
+
+\*******************************************************************/
+
+#include "../headers/buitin_libs.hs"
+
+extern void __ESBMC_atomic_begin();
+extern void __ESBMC_atomic_end();
+
```

Comment:
you probably don't need these extern declarations

Reply:
They are needed for compilation

Reply:
Compilation of what? ESBMC? c2goto?

These declarations are inserted by the preprocessor to every program.

Reply:
> Compilation of what? ESBMC? c2goto?

Boost Test and the Fuzzer

Reply:
why is it inside ESBMC source then? Should be moved to the unit test dir.

Reply:
Because `builtin_libs.c` is compiled as a library and then linked against the test

Reply:
If it is compiled using c2goto, it doesn't need the extern declaration. Please remove the extern declaration.